### PR TITLE
code cleanup: Remove impossible invoice condition

### PR DESCRIPTION
**Pull request by ilnytskyi**
### Description (*)

As the invoiceService declared to always return InvoiceInterface
```
$this->invoiceService->prepareInvoice($order, $invoiceItems);
```
in
https://github.com/ilnytskyi/magento2/blob/6df210e6e40c1e853137cf365bc390f6721d7011/app/code/Magento/Sales/Model/Service/InvoiceService.php#L148

This check is not needed
```
if (!$invoice)
```

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. create invoice
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Sales/Controller/Adminhtml/Order/Invoice/Save.php [hunk 1]
```diff
@@ -166,10 +166,6 @@ public function execute()
 
             $invoice = $this->invoiceService->prepareInvoice($order, $invoiceItems);
 
-            if (!$invoice) {
```
Comment (BarnyShergold) [inline app/code/Magento/Sales/Controller/Adminhtml/Order/Invoice/Save.php:169]: Won't this cause an issue now with the testExecuteNotValidPost() test? - need to see

**Comment by ilnytskyi** [comment]
@magento run all tests

**Review by Den4ik — CHANGES_REQUESTED** [review]
@ilnytskyi Please fix failed static test

**Comment by ilnytskyi** [comment]
@magento run all tests

**Comment by ilnytskyi** [comment]
@magento run all tests

**Comment by ilnytskyi** [comment]
@magento run Integration Tests

**Comment by ilnytskyi** [comment]
@magento run Functional Tests B2B

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-9220](https://jira.corp.magento.com/browse/ENGCOM-9220) has been created to process this Pull Request

**Review by Den4ik — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9220](https://jira.corp.magento.com/browse/ENGCOM-9220) has been created to process this Pull Request

**Review by BarnyShergold — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @BarnyShergold, thank you for the review.
[ENGCOM-9220](https://jira.corp.magento.com/browse/ENGCOM-9220) has been created to process this Pull Request

**Comment by ilnytskyi** [comment]
up

**Comment by ilnytskyi** [comment]
up

**Comment by andrewbess** [comment]
This PR is already merged

**Pull request merged**